lib/signex.ex

```diff
@@ -1,9 +1,53 @@
 defmodule SignEx do
-  require SignEx.Signer
-  require SignEx.Verifier
-  require SignEx.Helper
+  import SignEx.Helper
 
-  def sign(message, private_key, public_key), do: SignEx.Signer.sign(message, private_key, public_key)
-  def verify(message, signature, public_key), do: SignEx.Verifier.verify(message, signature, public_key)
-  def signature_params(signature), do: SignEx.Helper.signature_params(signature)
+  def sign(body, metadata, keypair) do
+    metadata = Map.merge(metadata, %{"digest" => generate_digest(body)})
+    parameters = SignEx.Signer.sign(metadata, keypair)
+    {:ok, {metadata, parameters}}
+  end
+
+  def signature_valid?(
+    headers,
+    params = %SignEx.Parameters{algorithm: algorithm},
+    public_key) when is_binary(public_key) and algorithm in ["rsa-sha512", "ec-sha512"] do
+    case Base.decode64(params.signature) do
+      {:ok, signature} ->
+        case fetch_keys(headers, params.headers) do
+          {:ok, ordered_headers} ->
+            signing_string = compose_signing_string(ordered_headers)
+            :public_key.verify(signing_string, :sha512, signature, decode_key(public_key))
```

Comment:
Isn't the use of atoms more or less for constants? or are you suggesting renaming it? e.g. `@digest_type :sha512`